src/account/operations/transaction/high_level/create_alias.rs

```diff
@@ -0,0 +1,169 @@
+    /// println!(
+    ///     "Transaction: {} Block sent: http://localhost:14265/api/core/v2/blocks/{}",
+    ///     transaction.transaction_id,
+    ///     transaction.block_id.expect("no block created yet")
+    /// );
+    /// ```
+    pub async fn create_alias_output(
+        &self,
+        alias_output_options: Option<AliasOutputOptions>,
+        options: Option<TransactionOptions>,
+    ) -> crate::Result<Transaction> {
+        let prepared_transaction = self.prepare_create_alias_output(alias_output_options, options).await?;
+        self.sign_and_submit_transaction(prepared_transaction).await
+    }
+
+    pub(crate) async fn prepare_create_alias_output(
+        &self,
+        alias_output_options: Option<AliasOutputOptions>,
+        options: Option<TransactionOptions>,
+    ) -> crate::Result<PreparedTransactionData> {
+        log::debug!("[TRANSACTION] prepare_create_alias_output");
+        let rent_structure = self.client.get_rent_structure().await?;
+
+        let controller_address = match alias_output_options
+            .as_ref()
+            .and_then(|options| options.address.as_ref())
+        {
+            Some(bech32_address) => Address::try_from_bech32(bech32_address)?.1,
+            None => {
+                self.public_addresses()
+                    .await
+                    .first()
+                    .ok_or(Error::FailedToGetRemainder)?
+                    .address
+                    .inner
+            }
+        };
+
+        let mut alias_output_builder =
+            AliasOutputBuilder::new_with_minimum_storage_deposit(rent_structure, AliasId::null())?
+                .with_state_index(0)
+                .with_foundry_counter(0)
+                .add_unlock_condition(UnlockCondition::StateControllerAddress(
+                    StateControllerAddressUnlockCondition::new(controller_address),
+                ))
+                .add_unlock_condition(UnlockCondition::GovernorAddress(GovernorAddressUnlockCondition::new(
+                    controller_address,
+                )));
+        if let Some(immutable_metadata) = alias_output_options
+            .as_ref()
+            .and_then(|options| options.immutable_metadata.clone())
+        {
+            alias_output_builder = alias_output_builder
+                .add_immutable_feature(Feature::Metadata(MetadataFeature::new(immutable_metadata)?));
+        };
+        if let Some(metadata) = alias_output_options.and_then(|options| options.metadata) {
+            alias_output_builder = alias_output_builder.add_feature(Feature::Metadata(MetadataFeature::new(metadata)?));
+        };
+
+        let outputs = vec![alias_output_builder.finish_output()?];
+
+        self.prepare_transaction(outputs, options).await
+    }
+
+    /// Get an existing alias output
+    pub(crate) async fn get_alias_output(&self, alias_id: Option<AliasId>) -> Option<(AliasId, OutputData)> {
```

Comment:
The option is there to get either the first one or a specific one ?

Reply:
Yes